### PR TITLE
arch/xtensa/esp32: move the esp_err.h to folder arch/chip, remove the same file in esp_idf_port;

### PR DESCRIPTION
**Pull request by glb1013**
There are two esp_err.h, so remove the one in the esp_idf_port/include. And the #ifndef macro to avoid multi include.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f036a2c2bcb59ef803490ba6f15eb8bc562c2421] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f036a2c2bcb59ef803490ba6f15eb8bc562c2421] - Code Rule Check OK.

File: external/esp_idf_port/esp32/dport_access.c [hunk 1]
```diff
@@ -41,7 +41,7 @@
 #include <string.h>
 
 #include "esp_attr.h"
-#include "esp_err.h"
+#include <arch/chip/esp_err.h>
 //#include "esp_intr.h"
```
Comment (NilabhShant) [inline external/esp_idf_port/esp32/dport_access.c]: Please remove it if not required.

File: os/arch/xtensa/src/esp32/chip/esp32_rtc_touch_pad.h [hunk 2]
```diff
@@ -36,7 +36,7 @@
 extern "C" {
 #endif
 //#include "esp_intr.h"
-//#include "esp_err.h"
+//#include <arch/chip/esp_err.h>
```
Comment (NilabhShant) [inline os/arch/xtensa/src/esp32/chip/esp32_rtc_touch_pad.h]: same as above.

Comment (xiarihf) [inline os/arch/xtensa/src/esp32/chip/esp32_rtc_touch_pad.h]: useless modification here.

File: external/esp_idf_port/include/esp_wifi_internal.h [hunk 3]
```diff
@@ -112,7 +112,7 @@ typedef enum {
  * @return
  *    - ESP_OK: succeed
  *    - ESP_ERR_NO_MEM: out of memory
- *    - others: refer to error code esp_err.h
+ *    - others: refer to error code <arch/chip/esp_err.h>
```
Comment (xiarihf) [inline external/esp_idf_port/include/esp_wifi_internal.h]: Please do not modify this line.

File: external/esp_idf_port/spi_flash/include/esp_partition.h [hunk 4]
```diff
@@ -36,7 +36,7 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <stddef.h>
-//#include "esp_err.h"
+#include <arch/chip/esp_err.h>
```
Comment (xiarihf) [inline external/esp_idf_port/spi_flash/include/esp_partition.h]: why include esp_err.h here, while it was commented in the old file?

File: external/esp_idf_port/esp32/dport_access.c [hunk 5]
```diff
@@ -41,7 +41,7 @@
...
 #include "rom/ets_sys.h"
 //#include "rom/uart.h"
```
Comment (ChinaTizenRT) [inline external/esp_idf_port/esp32/dport_access.c]: pls remove comment code //#include "rom/uart.h"

File: external/esp_idf_port/esp32/dport_access.c (hunk 1 above)
Comment (glb1013) [reply external/esp_idf_port/esp32/dport_access.c]: Removed.

File: os/arch/xtensa/src/esp32/chip/esp32_rtc_touch_pad.h (hunk 2 above)
Comment (glb1013) [reply os/arch/xtensa/src/esp32/chip/esp32_rtc_touch_pad.h]: Removed.

Comment (glb1013) [reply os/arch/xtensa/src/esp32/chip/esp32_rtc_touch_pad.h]: changed back.

File: external/esp_idf_port/include/esp_wifi_internal.h (hunk 3 above)
Comment (glb1013) [reply external/esp_idf_port/include/esp_wifi_internal.h]: changed back.

File: external/esp_idf_port/spi_flash/include/esp_partition.h (hunk 4 above)
Comment (glb1013) [reply external/esp_idf_port/spi_flash/include/esp_partition.h]: Removed

File: external/esp_idf_port/esp32/dport_access.c (hunk 5 above)
Comment (glb1013) [reply external/esp_idf_port/esp32/dport_access.c]: Removed

**Comment by seinfra** [comment]
Target : [fec260344dc342d54ffda10e978994f04b16d178] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fec260344dc342d54ffda10e978994f04b16d178] - Code Rule Check OK.

**Pull request merged**